sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1867,7 +1867,7 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
   def identifierHavingMoreThanTwoNamePartsError(
       quoted: String, identifier: String): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1185",
+      errorClass = "IDENTIFIER_TOO_MANY_NAME_PARTS",
       messageParameters = Map(
         "quoted" -> quoted,
```

Comment:
Please, pass the original parts and use `toSQLId()` to always quote them by backticks.

Reply:
Done, please check